apriltags_ros/src/apriltag_detector.cpp

```diff
@@ -473,25 +514,168 @@ tf::Transform AprilTagDetector::getDepthImagePlaneTransform(const sensor_msgs::P
   clipPolygon.push_back(pcl::PointXYZ(polygon[0].first, polygon[0].second, 0));
 
-  for (int x = 0; x < pointCloud->width; x++)
+  // generate the transform from depth_camera_reference frame to rgb_camera_reference frame
+  tf::Transform depth_to_rgb_transform;
+  string rgb_camera_frame_name = rgb_info->header.frame_id;
+  string depth_camera_frame_name = cloud->header.frame_id;
+
+  ROS_WARN("HERE %d %s name of the rgb_camera frame %s name of the depth camera frame %s", __LINE__, __FILE__, rgb_camera_frame_name.c_str(), depth_camera_frame_name.c_str());
+
+  getTransform(rgb_camera_frame_name, depth_camera_frame_name, depth_to_rgb_transform);
+
+  // generate a projection from rgb_camera_reference frame to image frame that is the P matrix in the rgb_info object
+
+  // actually convert the whole point cloud from depth frame to rgb frame
+
+  pcl::PointCloud<pcl::PointXYZRGB> point_cloud_in_rgb_frame;
+
+  //long double Bx = depth_info->P[3]/(-1 * depth_info->P[0]);
+  //ROS_WARN("HERE %d %s Bx = %f", __LINE__, __FILE__, Bx);
+  //long double By = 0;//depth_info->P[7]/-depth_info->P[1];
+
+  //depth_to_rgb_transform.setIdentity();
+  //depth_to_rgb_transform.setOrigin(tf::Vector3{Bx,By,0});
+  pcl_ros::transformPointCloud(*pointCloud, point_cloud_in_rgb_frame, depth_to_rgb_transform);
+
+    //std::for_each(pointCloud->begin(), pointCloud->end(), [&depth_to_rgb_transform](pcl::PointXYZRGB & point) -> void {
+    //tf::Vector3 pcl_point(point.x, point.y, point.z); pcl_point = depth_to_rgb_transform * pcl_point; point.x = pcl_point.x(); point.y = pcl_point.y(); point.z = pcl_point.z(); });
+
+
+  //std::for_each(pointCloud->begin(), pointCloud->end(), [Bx,By, &point_cloud_in_rgb_frame](pcl::PointXYZRGB & point) -> void {
+  //      pcl::PointXYZRGB pcl_point(point.x+Bx, point.y+By, point.z);
+  //      point_cloud_in_rgb_frame.push_back(pcl_point);
+  //});
+  point_cloud_in_rgb_frame.header.frame_id = rgb_camera_frame_name;
+  //depth_to_rgb_transform
+    //
+    Eigen::Matrix3f projection_mat;
+    projection_mat << rgb_info->P[0], rgb_info->P[1], rgb_info->P[2], rgb_info->P[4], rgb_info->P[5], rgb_info->P[6], rgb_info->P[8], rgb_info->P[9], rgb_info->P[10];
+
+    for (int i = 0; i != 12; ++i) {
+        ROS_WARN("HERE %d %s rgb_info->p[%d] = %f", __LINE__, __FILE__, i, rgb_info->P[i]);
+    }
+
+    for (int i = 0; i != 12; ++i) {
+        ROS_WARN("HERE %d %s depth_info->p[%d] = %f", __LINE__, __FILE__, i, depth_info->P[i]);
+    }
+  ROS_DEBUG_THROTTLE(5.0, "about to analyze the cloud");
+  ROS_WARN("HERE %d %s about to analyze the cloud", __LINE__, __FILE__);
+
+
+    float minX = 10000;
+    float minY = 10000;
+    float maxX = -10000;
+    float maxY = -10000;
+
+  float minXrgb = 10000;
+  float minYrgb = 10000;
+  float maxXrgb = -10000;
+  float maxYrgb = -10000;
+
+  double fx;
+  double fy;
+  double cx;
+  double tx = 0;
+  double cy;
+  double ty = 0;
+  if (true) {
```

Comment:
I think you are correct that we should use the same logic as the other section of the april tag code but we should use the same boolean argument instead (see ^^^).